### PR TITLE
feat(handle): add selected property and calciteHandleChange event.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #8522

## Summary

- depends on #8483
- Renames `activated` property to `selected` to align with other components
- Makes `selected` public
- Adds public event `calciteHandleChange` for when `selected` is changed via user
- Renames internal event `calciteInternalHandleChange` to `calciteInternalAssistiveTextChange` to better illustrate its purpose
- adds tests

### CODE REVIEW

File: packages/calcite-components/src/components/handle/handle.tsx [hunk 1]
```diff
@@ -180,14 +180,21 @@ export class Handle implements LoadableComponent, T9nComponent, InteractiveCompo
   // --------------------------------------------------------------------------
 
   /**
-   * Emitted when the handle is activated and the up or down arrow key is pressed.
+   * Emits whenever the component is selected or unselected.
```
Comment (benelan) [inline packages/calcite-components/src/components/handle/handle.tsx:183]: @geospatialem @DitwanP do we have a convention on "Emits" vs "Emitted"? I've seen them both throughout the code base. Maybe we should pick one and do some cleanup in a different PR? I'm partial to "Emits" because we use "Specifies" for properties.

**Review by benelan — APPROVED** [review]
LGTM! We already have `selected` in other components so it's safe addition, naming-wise. Otherwise I'd say we should wait for @jcfranco to get back.

File: packages/calcite-components/src/components/handle/handle.tsx (hunk 1 above)
Comment (geospatialem) [reply packages/calcite-components/src/components/handle/handle.tsx:183]: Good question, we've moved towards "Emits" for the reason you specify above, to keep present tense on the docs. It looks like there are two outliers in `list` and `list-item-group`. Otherwise there is mention of "emitted" throughout the code, but not in the doc. Added this note to our conventions for #7071.

**Comment by driskull** [comment]
@jcfranco when you're back. let me know if we should revert this or keep it internal.
